### PR TITLE
Updated instructions for working with WSL and newer tools

### PR DESCRIPTION
**Pull request by mhammond**
Flagging Ryan for the WSL changes and Ed to see if there's anything obvious that's not going to work elsewhere...

### CODE REVIEW

File: docs/howtos/setup-android-build-environment.md [hunk 1]
```diff
@@ -108,10 +85,106 @@ you will find:
...
+- `ANDROID_NDK_API_VERSION`
+- `ANDROID_HOME`
+- `JAVA_HOME`
+
+These variables are required every time you build, so you should add them to
+a rc file or similar so they persist between reboots etc.
+
+1. Install the SDK, largely as described above.
+
+2. Install NDK r21 from https://developer.android.com/ndk/downloads
```
Comment (rfk) [inline docs/howtos/setup-android-build-environment.md:158]: I'm surprised that the instructions above seemed to install r20, but these outdated instructions call for r21. I guess it's related to the build error you mention below, but OTOH, I'm definitely using r21 and it seems to be working OK.

File: docs/howtos/setup-android-build-environment.md [hunk 2]
```diff
@@ -128,43 +201,35 @@ You will need the following tools in WSL:
 
 * unzip - `sudo apt install unzip`
 
-* python 3 - `sudo apt install python3`
+* python 3 - `sudo apt install python3` (XXX - TODO - you actually need 3.6
+  and if your distro doesn't some with that, you need to google how to make it
```
Comment (rfk) [inline docs/howtos/setup-android-build-environment.md:205]: s/some/come/

File: docs/howtos/setup-android-build-environment.md [hunk 3]
```diff
@@ -177,3 +242,23 @@ We now want to configure maven to use the native windows maven repository - then
...
 
 \o/
+
+# Other obsolete content
+
+This content is probably outdated. It should either be removed, or saved!
+
+## Publishing to your local maven repo
+
+Note that this has been make obsolete by the `autoPublish` mechanisms above.
```
Comment (rfk) [inline docs/howtos/setup-android-build-environment.md:252]: s/make/made/

File: docs/howtos/setup-android-build-environment.md [hunk 4]
```diff
@@ -177,3 +242,23 @@ We now want to configure maven to use the native windows maven repository - then
 * Now you should be ready to roll - `./gradlew install` should complete and publish the components to your native maven repo!
 
 \o/
+
+# Other obsolete content
+
+This content is probably outdated. It should either be removed, or saved!
```
Comment (rfk) [inline docs/howtos/setup-android-build-environment.md:248]: I would support just removing this FWIW; if you already know about `publishToMavenLocal` then you don't need these docs, and if you don't know about `publishToMavenLocal` then you're near certain to have a bad time and should instead use the auto-publish flow. I believe the detailed docs on the autopublish flow also have fallback docs describing `publishToMavenLocal`, so we wouldn't be losing much by removing them here.

**Comment by mhammond** [comment]
TBH, I now have a working environment, so I'm not that interested in starting from scratch again. Given that, it seems wasteful to have 2 people work on making this document match how docker does things when 1 would work just as well - I guess you have vmware and a windows license, or can obtain one from servicenow.

I'm happy to close this if you think it's a step backwards from the old version, but that old version simply did not work - let me know if I should just close it.

**Comment by mhammond** [comment]
(well, I expect you don't need windows - a clean ubuntu 18 seems fine?)

File: docs/howtos/setup-android-build-environment.md [hunk 5]
```diff
@@ -108,10 +85,106 @@ you will find:
...
+
+  If not already installed, you can install the "command-line" only tools:
+
+   - Visit https://developer.android.com/studio/, at the bottom of the page locate the current SDKs for linux
+     at time of writing, this is https://dl.google.com/android/repository/commandlinetools-linux-6514223_latest.zip
+
+  Install it by executing:
+
+  > % cd ~  
+  > % mkdir -p android-sdk/cmdline-tools
```
[the remark this answers is not in the document]
Comment (eoger) [reply docs/howtos/setup-android-build-environment.md:110]: Turns out this is how one installs the SDK these days

File: docs/howtos/setup-android-build-environment.md [hunk 6]
```diff
@@ -108,10 +85,106 @@ you will find:
...
+
+  > `$ANDROID_HOME/cmdline-tools/tools/bin/sdkmanager --install "ndk;20.0.5594570"
+
+   - If you have Android Studio or otherwise a GUI version of the SDK, use that
+     GUI.
+
+  In either case, you don't need to select where it is installed or to set
+  further environment variables.
+
+At the end of this process, all you need is `ANDROID_HOME` set!
```
[the remark this answers is not in the document]
Comment (eoger) [reply docs/howtos/setup-android-build-environment.md:137]: Turns out my NDK knowledge was out-of-date, and installing using the SDK is The Proper Way now. Thanks Mark for digging into this, I'll make a PR so we use this everywhere (it should make new contributors lives easier).

**Comment by rfk** [comment]
Let's close this in favour of https://github.com/mozilla/application-services/pull/3228

**Pull request closed without merging**